src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponentSpec.scala

```diff
@@ -398,4 +400,63 @@ class ClusterComponentSpec extends TestComponent with FlatSpecLike with CommonTe
     dbFutureValue { _.clusterQuery.getByGoogleId(c1.googleId) } shouldBe Some(updatedC1Again)
   }
 
+  it should "get list of clusters to auto freeze" in isolatedDbTest {
+    val c1 = Cluster(
+      clusterName = name1,
+      googleId = UUID.randomUUID(),
+      googleProject = project,
+      serviceAccountInfo = ServiceAccountInfo(Some(serviceAccountEmail), Some(serviceAccountEmail)),
+      machineConfig = MachineConfig(Some(0), Some(""), Some(500)),
+      clusterUrl = Cluster.getClusterUrl(project, name1, clusterUrlBase),
+      operationName = OperationName("op1"),
+      status = ClusterStatus.Running,
+      hostIp = Some(IP("numbers.and.dots")),
+      creator = userEmail,
+      createdDate = Instant.now(),
+      destroyedDate = None,
+      labels = Map("bam" -> "yes", "vcf" -> "no"),
+      jupyterExtensionUri = None,
+      jupyterUserScriptUri = None,
+      Some(GcsBucketName("testStagingBucket1")),
+      List.empty,
+      Set(masterInstance),
+      None,
+      Instant.now()
+    )
+
+    val c2 = Cluster(
+      clusterName = name2,
+      googleId = UUID.randomUUID(),
+      googleProject = project,
+      serviceAccountInfo = ServiceAccountInfo(None, Some(serviceAccountEmail)),
+      machineConfig = MachineConfig(Some(0),Some(""), Some(500)),
+      clusterUrl = Cluster.getClusterUrl(project, name2, clusterUrlBase),
+      operationName = OperationName("op2"),
+      status = ClusterStatus.Stopped,
+      hostIp = None,
+      creator = userEmail,
+      createdDate = Instant.now(),
+      destroyedDate = None,
+      labels = Map.empty,
+      jupyterExtensionUri = Some(jupyterExtensionUri),
+      jupyterUserScriptUri = Some(jupyterUserScriptUri),
+      Some(GcsBucketName("testStagingBucket2")),
+      List.empty,
+      Set.empty,
+      None,
+      Instant.now()
+    )
+
+    dbFutureValue { _.clusterQuery.save(c1, gcsPath("gs://bucket1"), Some(serviceAccountKey.id)) } shouldEqual c1
+    dbFutureValue { _.clusterQuery.save(c2, gcsPath("gs://bucket1"), Some(serviceAccountKey.id)) } shouldEqual c2
+
+    dbFutureValue { _.clusterQuery.getClustersReadyToAutoFreeze(autoFreezeconfig.autoFreezeAfter) } shouldBe List.empty
+
+    eventually(timeout(Span(30, Seconds))) {
+      val autoFreezeList = dbFutureValue { _.clusterQuery.getClustersReadyToAutoFreeze(autoFreezeconfig.autoFreezeAfter) }
+      autoFreezeList should contain (c1)
+      //c2 is already stopped
+      autoFreezeList should not contain c2
```

Comment:
Ideally, we would also test the other stoppable and unstoppable statuses as well. I'm wary of creating more duplicate code with `c3`, `c4`, etc., though as they caused a lot of code updates when I was updating the `Cluster` `case class`. It'd be great to have a `def getCluster: Cluster` (perhaps in `CommonTestData`) that yields the most commonly used configuration and every other `cX` would do something like 

```scala
val c2 = getCluster.copy(name = "cluster2", status = "Updating", etc.)
```

I know this is not something introduced by this PR and shouldn't be a blocker but I'd be happy to add this if you like...just let me know.

Reply:
+1 for this. I'll create a housekeeping ticket for this.

Reply:
+1 on creating common cluster instances/helper methods that can be reused across tests. There is currently a lot of repeated boilerplate in test setup code.

Reply:
https://github.com/DataBiosphere/leonardo/issues/394